pkg/executor/show.go

```diff
@@ -717,6 +723,13 @@ func (e *ShowExec) fetchShowColumns(ctx context.Context) error {
 		} else if fieldPatternsLike != nil && !fieldPatternsLike.DoMatch(col.Name.L) {
 			continue
 		}
+		if !passTblPrivCheck {
+			if !checker.RequestVerification(activeRoles, e.DBName.O, tb.Meta().Name.O, col.Name.O, priv) {
```

Comment:
How about use
```
if checker.RequestVerification(col) {
      passColPrivCheck = true
}
```
instead of 
```
if !checker.RequestVerification(col) {
    continue
}
```

Reply:
We need to skip the record if it lacks the column's privilege. So a `continue` is needed here.

Reply:
@CbcWestwolf Got it. I think adding a comment would be better.

Reply:
Sure, added in [`4fe5a54` (#61679)](https://github.com/pingcap/tidb/pull/61679/commits/4fe5a540468022828beed9364a3cd1b0ba001435)